universe/interface.go

```diff
@@ -552,6 +552,44 @@ type FederationLog interface {
 	LogNewSyncs(ctx context.Context, addrs ...ServerAddr) error
 }
 
+// ProofType is an enum that describes the type of proof which can be stored in
+// a given universe.
+type ProofType int
```

Comment:
Can use `uint8` to avoid different types being used on different CPU architectures.